linters/importorder/main.go

```diff
@@ -0,0 +1,373 @@
+		}
+	}
+}
+
+func handleImportPaths(importPaths []string, buildTags, patterns []string) lintErrors {
+	fs := token.NewFileSet()
+
+	ctx := build.Default
+	ctx.BuildTags = buildTags
+
+	conf := loader.Config{
+		Fset:  fs,
+		Build: &ctx,
+		// Since we are not concerned with the entire file, we should only parse the imports
+		ParserMode: parser.ImportsOnly,
+		// Continue even if type or IO errors are present
+		AllowErrors: true,
+		TypeChecker: types.Config{
+			Error: func(e error) {},
+		},
+	}
+
+	for _, importPath := range importPaths {
+		conf.ImportWithTests(importPath)
+	}
+
+	prog, err := conf.Load()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var groupedLintErrors lintErrors
+	for _, pkg := range prog.InitialPackages() {
+		for _, file := range pkg.Files {
+			imports := imports(fs, file)
+			if len(imports) == 0 {
+				continue
+			}
+			validateImportDecl(imports)
+			goldStandard, err := getGoldStandard(imports, patterns)
+			// for testing
+			// for _, g := range goldStandard.Groups {
+			// 	fmt.Println(g.Imports)
+			// }
+			// fmt.Println("********")
+			if err != nil {
+				groupedLintErrors = append(groupedLintErrors, lintError{err: err, fileName: fs.Position(file.Pos()).Filename})
+			} else {
+				if !compareImports(goldStandard, imports[0]) {
+					groupedLintErrors = append(groupedLintErrors, lintError{fileName: fs.Position(file.Pos()).Filename,
+						originalDecl: imports[0],
+						goldStandard: goldStandard,
+						err:          errOutOfOrder,
+					})
+				}
+			}
+		}
+	}
+	return groupedLintErrors
+}
+
+func compareImports(goldStandard, originalImportDecl importDecl) bool {
+	goldGroups := goldStandard.Groups
+	originalGroups := originalImportDecl.Groups
+
+	if len(goldGroups) != len(originalGroups) {
+		return false
+	}
+
+	for i, goldGroup := range goldGroups {
+		for j, goldImport := range goldGroup.Imports {
+			if goldImport != originalGroups[i].Imports[j] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func getGoldStandard(imports []importDecl, patterns []string) (importDecl, error) {
+	var emptyImportDecl importDecl
+	if len(imports) > 1 {
+		return emptyImportDecl, errMultipleImport
+	}
+
+	if len(imports) == 0 {
+		return emptyImportDecl, nil
+	}
+
+	combinedImports := concatenateImports(imports[0])
+	goldStandard := createGoldStandard(combinedImports, patterns)
+
+	return goldStandard, nil
+}
+
+func createGoldStandard(imports []importSpec, patterns []string) importDecl {
+	groups := make([]importGroup, 0, len(patterns))
+	importMap := convertToMap(imports)
+
+	for i, pattern := range patterns {
+		var tempGroup []importSpec
+		for imp := range importMap {
+			if pattern == standardImportGroup {
+				tempGroup = orderStdLib(tempGroup, imp, importMap)
+			}
+			if pattern == externalImportGroup {
+				tempGroup = orderExt(tempGroup, imp, importMap, patterns, i)
+			}
+			if strings.Contains(imp.Path, pattern) {
+				tempGroup = append(tempGroup, imp)
+				delete(importMap, imp)
```

Comment:
could you pull out the `delete(importMap, imp)` from each of the functions and do it in the loop, will be easier to trace through the logic that way, i.e.

```
func createGoldStandard(imports []importSpec, patterns []string) importDecl {
	groups := make([]importGroup, 0, len(patterns))
	importMap := convertToMap(imports)

	for i, pattern := range patterns {
		var tempGroup []importSpec
		for imp := range importMap {
                        switch {
                        case pattern == standardImportGroup:
				tempGroup = orderStdLib(tempGroup, imp, importMap)
                        case pattern == externalImportGroup:
				tempGroup = orderExt(tempGroup, imp, importMap, patterns, i)
			case  strings.Contains(imp.Path, pattern:
				tempGroup = append(tempGroup, imp)
			default:
				// can this ever happen?
		}
		delete(importMap, imp)

```